bfdd/bfd.c

```diff
@@ -997,9 +997,18 @@ static void bs_down_handler(struct bfd_session *bs, int nstate)
 		 */
 		bs->ses_state = PTM_BFD_INIT;
 
-		/* Answer peer with INIT immediately in passive mode. */
+		/*
+		 * RFC 5880, Section 6.1.
+		 * A system taking the Passive role MUST NOT begin
+		 * sending BFD packets for a particular session until
+		 * it has received a BFD packet for that session, and thus
+		 * has learned the remote system's discriminator value.
+		 *
+		 * Now we can start transmission timer in passive mode.
+		 */
 		if (CHECK_FLAG(bs->flags, BFD_SESS_FLAG_PASSIVE))
-			ptm_bfd_snd(bs, 0);
+			ptm_bfd_xmt_TO(bs, 0);
```

Comment:
This is the right change, but you have to disable the transmission when the session goes back down from init/up status.

Reply:
Yes, i had checked. Disable mechanism was already there for passive mode.

Reply:
It is done in ptm_bfd_sess_dn, but this function is only called when the session goes from Up to Down. There are many more state changes, when we must disable the timers and it's not done now, for example – Up->AdminDown, Init->Down, Init->AdminDown, maybe more. Please, check all of them.

Reply:
Hi only “to Down” is special for passive mode. ” to AdminDown” is same to active mode. Anyway i will check more carefully.

Reply:
@idryzhov I have double-check these conditions, it is safe.

As to destroy this xmttimer:

- Enable/noshutdown to Down: If passive mode, then do not start xmttimer(and recvtimer) (`bfd_session_enable/bfd_set_passive_mode`, ignore an working passive mode session)

- From Init/Up to Down: If passive mode, then remove xmttimer (and recvtimer)  (`ptm_bfd_sess_dn`)

- From Init/Up to AdminDown: For both passive and active mode, then remove xmttimer(and recvtmer) (`bfd_set_shutdown`)

Reply:
This is not true.
1. Let's look at `bs_init_handler`. If we receive a `PTM_BFD_ADM_DOWN` from neighbor, then the session goes down and the timer must be stopped. It is not done now.
2. Let's look at `bs_up_handler`. If we receive a `PTM_BFD_ADM_DOWN` from neighbor, then `bs_neighbour_admin_down_handler` is called, and it also must stop the timer. It is also not done now.
Please, fix this.

Reply:
Really missed the case of receiving `AdminDown` from neighbor, thanks for your correction.
Done in last push.